### DISCUSSION TITLE
Basic use appears to generate unusable output

### DISCUSSION THREAD

**Discussion by danielquinn** [coqui-ai/TTS#982] [General Q&A] [answered]
I wasn't sure how to title this, but following the basic instructions, I did the following:

```
$ pip install tts
$ tts --text hello
$ mpv tts_output.wav
```

The result is attached and... it's pretty bad.  I tried variations using different models:

```
$ tts --text hello --model_name=tts_models/en/ljspeech/tacotron2-DCA
$ tts --text hello --model_name=tts_models/en/ek1/tacotron2
$ tts --text hello --model_name=tts_models/en/ljspeech/glow-tts
```

The results for all of these are unusable.  Have I missed something?  The results seem really all over the place, with `hello` generating some creepy OOoooOOoOoOo noises, but `hello I am a sample text` sounding fabulous.  Unfortunately, my application needs some semblance of predictability, and it's conceivable that I'd want my program to "say hello" so this is causing me some pain :-(

I'm pretty experienced in software development, but have next to no experience with ML so "just train the model to do X" doesn't really mean anything to me.  If I shouldn't consider this library to be off-the-shelf text-to-speech, just let me know and I'll explore other options, but so far it's seemed remarkably user-friendly for an ML project, so I thought I'd give it a go as the voice for my current project.

### Samples

I had to zip these 'cause GitHub doesn't allow .wav files.

* [tts --text "hello"](https://github.com/coqui-ai/TTS/files/7621369/tts_output.zip)  (Bad)
* [tts --text "hello I am a sample text"](https://github.com/coqui-ai/TTS/files/7621379/tts_output.zip) (Great!)

**Comment by DelanoLeslie** [comment, marked as answer]
Try giving the input punctuation like "Hello." or "Hello, this is a sample."

I think this should help. A majority of the time the datasets these models are trained use punctuation, so the lack of punctuation may confuse it.

Upon further testing, it looks like punctuation does fix the issue.

[tts --text "hello"](https://github.com/coqui-ai/TTS/files/7622596/tts_output.2.zip)
[tts --text  "Hello."](https://github.com/coqui-ai/TTS/files/7622599/tts_output.3.zip)

**Reply by danielquinn** [reply]
Ooh, that's a good to know.  It might be a good idea to include that sort of thing in the README though as it's likely that people wouldn't think of this by default.

**Comment by erogol** [comment]
You need to add punctuations since all the models are trained with punctuated sentences. 

I close this as it is not a bug

**Comment by saona-raimundo** [comment]
Would it not be a better solution to "pre-process" input text?

In the pre-processing, one could add "missing punctuation" (and potentially other things) to make any input closer to the trained sentences and ultimately give better results(?)

For this case, it was as simple as adding `.` at the end of the sentence, a procedure that can be made totally generic ovr input!

**Reply by erogol** [reply]
We don't always know what every model expects to see.  So such hard-coded pre-processing might cause obfuscated bugs.

So again our tools are mostly for experimenting with the models not intended for deployment. If you deploy your own model, it is up to the dev to deal with these details.

**Reply by saona-raimundo** [reply]
Thank you for the answer! I assumed this could have been model-related, but I agree that such pre-processing would involve the model and the data, and therefore it is better suited as part of a finished product than to experiments!

**Comment by jeffrafter** [comment]
I found this post while trying to find the answer. I've tried a number of ways to get `TTS` running on an Apple Mac M1 including running on Docker with https://github.com/synesthesiam/coqui-docker. Using version 0.5.0 and the latest models I am also getting unusable about - even with punctuation. I am sure I am doing something wrong:

./tts --text "Hello, this is Mike."

https://user-images.githubusercontent.com/4064/148230827-a1bfe13e-b06a-4bbe-a1e6-59a706f69b6d.mp4

./tts --text "Hello, this is mike. " --model_name tts_models/en/ljspeech/speedy-speech-wn

https://user-images.githubusercontent.com/4064/148231213-8c720522-7d50-4ee9-a03a-d05487415991.mp4

Is there a requirement on the machine or memory?

**Reply by saona-raimundo** [reply]
Interesting! This might imply that it is not only punctuation but also capitalization that has an impact on the output quality?

In my fresh install of tts (in Windows), I do not see your problems. The quality is much better.
Details:
> Using model: Tacotron2
 > Model's reduction rate `r` is set to: 1
 > Vocoder Model: hifigan
 > Generator Model: hifigan_generator
 > Discriminator Model: hifigan_discriminator

The difference between capitalization is heard in the rhythm of the voice, which is nice :)
